packages/headless/src/features/facets/category-facet-set/category-facet-set-slice.ts

```diff
@@ -58,7 +58,11 @@ export const categoryFacetSetReducer = createReducer(
         Object.keys(state).forEach((id) => {
           const request = state[id]!.request;
           const path = cf[id] || [];
-          selectPath(request, path, request.numberOfValues);
+          const numberOfValues = state[id]!.initialNumberOfValues;
+          if (!path.length) {
+            request.numberOfValues = numberOfValues;
+          }
```

Comment:
Could you give us some context behind these changes? and if it is specific to the `restoreSearchParameters` case?
The `selectPath` function is used in a few places, so maybe that needs to be adjusted?

Reply:
It's not an issue with selectPath, it seems to work fine

> const numberOfValues = state[id]!.initialNumberOfValues;

E.g. I'm drilling down to the last child, on the request the numberOfValues will then be 1 since 1 element is returned from the API. When doing "back" I'm now going back a level up where there was 5 values, but only 1 is requested. This is why I always reset to the initial number of values, I want all available values.

> if (!path.length) {
>     request.numberOfValues = numberOfValues;
> }
If i'm going back to the top level, I want to initialNumberOfValues apply it to the request directly as well

Reply:
Hmm, I'm not sure, I feel this bug is an artifact from when I refactored to store the initial number of values in state.
Shouldn't the third argument to `selectPath` always be the initial number of values?

```
export function selectPath(
  request: CategoryFacetRequest,
  path: string[],
  retrieveCount: number
) {
  request.currentValues = buildCurrentValuesFromPath(path, retrieveCount);
  request.numberOfValues = path.length ? 1 : request.numberOfValues;
  ...
}
```
would become

```
export function selectPath(
  request: CategoryFacetRequest,
  path: string[],
  initialNumberOfValues: number
) {
  request.currentValues = buildCurrentValuesFromPath(path, initialNumberOfValues);
  request.numberOfValues = path.length ? 1 : initialNumberOfValues;
  ...
}
```

and the `retrieveCount` param on `selectCategoryFacetSearchResult` would essentially be deprecated?

Reply:
I see!